tests/e2e/e2e_test.go

```diff
@@ -171,16 +173,61 @@ func (s *IntegrationTestSuite) TestTWAP() {
 	poolId := chainANode.CreatePool(poolFile, initialization.ValidatorWalletName)
 	swapWalletAddr := chainANode.CreateWallet(walletName)
 
+	timeBeforeSwap := chainANode.QueryLatestBlockTime()
+	// Wait for the next height so that the requested twap
+	// start time (timeBeforeSwap) is not equal to the block time.
+	chainA.WaitForNumHeights(1)
+
+	s.T().Log("querying for the first TWAP to now before swap")
+	twapFromBeforeSwapToBeforeSwapOne, err := chainANode.QueryGetArithmeticTwapToNow(poolId, denomOne, denomTwo, timeBeforeSwap)
+	s.Require().NoError(err)
+
 	chainANode.BankSend(coinIn, chainA.NodeConfigs[0].PublicAddress, swapWalletAddr)
-	heightBeforeSwap := chainANode.QueryCurrentHeight()
 
-	// Triggers the creation of TWAP records.
-	chainANode.SwapExactAmountIn(coinIn, minAmountOut, fmt.Sprintf("%d", poolId), denomOut, swapWalletAddr)
+	s.T().Log("querying for the second TWAP to now before swap, must equal to first")
+	twapFromBeforeSwapToBeforeSwapTwo, err := chainANode.QueryGetArithmeticTwapToNow(poolId, denomOne, denomTwo, timeBeforeSwap.Add(50*time.Millisecond))
+	s.Require().NoError(err)
+
+	// Since there were no swaps between the two queries, the TWAPs should be the same.
+	s.Require().Equal(twapFromBeforeSwapToBeforeSwapOne, twapFromBeforeSwapToBeforeSwapTwo)
+
+	s.T().Log("performing swap")
+	chainANode.SwapExactAmountIn(coinIn, minAmountOut, fmt.Sprintf("%d", poolId), denomTwo, swapWalletAddr)
 
 	keepPeriodCountDown := time.NewTimer(initialization.TWAPPruningKeepPeriod)
 
-	// Make sure still producing blocks.
-	chainA.WaitUntilHeight(heightBeforeSwap + 3)
+	// Make sure that we are still producing blocks and move far enough for the swap TWAP record to be created
+	// so that we can measure start time post-swap (timeAfterSwap).
+	chainA.WaitForNumHeights(2)
+
+	// Measure time after swap and wait for a few blocks to be produced.
+	// This is needed to ensure that start time is before the block time
+	// when we query for TWAP.
+	timeAfterSwap := chainANode.QueryLatestBlockTime()
+	chainA.WaitForNumHeights(2)
+
+	// TWAP "from before to after swap" should be different from "from before to before swap"
+	// because swap introduces a new record with a different spot price.
+	s.T().Log("querying for the TWAP from before swap to now after swap, must not equal to TWAPs before swap")
+	twapFromBeforeSwapToAfterSwap, err := chainANode.QueryGetArithmeticTwapToNow(poolId, denomOne, denomTwo, timeBeforeSwap)
+	s.Require().NoError(err)
+	s.Require().NotEqual(twapFromBeforeSwapToBeforeSwapOne, twapFromBeforeSwapToAfterSwap)
```

Comment:
While I don't think we should be checking for exact values, I do think NotEqual might be slightly too broad here (for example, a nil record is not equal but probably not what we want to see). I think a safe middle ground would just be doing some napkin math and at least asserting one should be GT or LT the other. Thoughts?

Reply:
In my opinion, the exact math should be tested at a lower level of abstraction. In e2e, we are just checking that system components work well together without diving deep into the calculation details.

Also, this is not 100% deterministic since this is twap to now where blocks and their respective time might slightly vary between runs

Reply:
I agree that the exact math should not be tested here. I guess what I am saying is `NotEqual` will return True if the twap record we retrieve is nil. We should for sure know which twap (before or after swap) is greater than the other, so a more specific `s.Require().True(twapFromBeforeSwapToBeforeSwapOne.GT(twapFromBeforeSwapToAfterSwap))` (or whatever the properly verbiage is) might be more desirable. 

This is not PR blocking at all, just a suggestion that may save us from missing issues in the future.

Reply:
Will try to address this before merging. Marking this as draft for now